week3/project/ecommerce/src/App.js

```diff
@@ -0,0 +1,38 @@
+import './App.css';
+import { Routes, Route } from 'react-router-dom'
+import ProductsPage from './pages/ProductsPage/ProductsPage'
+import SingleProductPage from './pages/SingleProductPage/SingleProductPage'
+import NotFoundPage from './pages/NotFoundPage/NotFoundPage';
+import HomePage from './pages/HomePage/HomePage';
+import FavoriteItemsPage from './pages/FavoriteItemsPage/FavoriteItemsPage';
+import { FavoriteIdItemContextProvider } from './contexts/FavoriteIdItemContext'
+
+
+
+function App() {
+
+   return (
+      <div className="App">
+         <Routes>
+            <Route path='/' element={<HomePage />} />
+            <Route path='/products' element={
+               <FavoriteIdItemContextProvider>
+                  <ProductsPage />
+               </FavoriteIdItemContextProvider>
+            } />
+            <Route path='/products/product/:itemId' element={
+               <FavoriteIdItemContextProvider>
+                  <SingleProductPage />
+               </FavoriteIdItemContextProvider>
+            } />
+            <Route path='/products/favorite' element={
+               <FavoriteIdItemContextProvider>
+                  < FavoriteItemsPage />
+               </FavoriteIdItemContextProvider>
+            } />
+            <Route path='*' element={<NotFoundPage />} />
+         </Routes>
```

Comment:
you have here a problem that if you go back to the home page it will erase your fav list .. and you can fix this by wrapping the `Routes` by the `FavoriteContext`

✏️need to modify✏️